nengo_loihi/builder/connection.py

```diff
@@ -369,10 +369,10 @@ def build_solver(model, solver, conn, rng, sampled_transform):
 
 @Builder.register(NoSolver)
 def build_no_solver(model, solver, conn, rng, sampled_transform):
+    args = (model, solver, conn, rng)
     if nengo_transforms is None:
-        return _build_no_solver(model, solver, conn, rng, sampled_transform)
-    else:
-        return _build_no_solver(model, solver, conn, rng)
+        args += (sampled_transform,)
```

Comment:
How does this work with tuples?? I thought they were immutable. I guess here `y += x` is just shorthand for `y = y + x`, which is very strange to me. (If you do `a = (1,2,3); b = a; a += (4,); print(b)`, it prints "(1, 2, 3)", which is different than the `[1, 2, 3, 4]` you get if `a` is a list.)

Anyway, I guess it's fine, I just had never seen that before.

Reply:
They are immutable yeah. Under the hood, `+=` results in a call to the left hand side's `__iadd__` method, which can do whatever it wants. I guess for lists, it does the append in-place and for tuples it makes a new tuple and returns the new instance.